clients/drcachesim/analysis_tool.h

```diff
@@ -277,6 +283,37 @@ template <typename RecordType> class analysis_tool_tmpl_t {
     {
         return nullptr;
     }
+    /**
+     * Finalizes the interval snapshots in the given \p interval_snapshots list.
+     * This callback provides an opportunity for tools to make any holistic
+     * adjustments to the snapshot list now that we have all of them together. This
+     * may include, for example, computing the diff with the previous snapshot.
+     *
+     * Tools can modify the individual snapshots and also the list of snapshots itself.
+     * If some snapshots are removed, release_interval_snapshot() will not be invoked
+     * for them and the tool is responsible to de-allocate the resources. Adding new
+     * snapshots to the list is undefined behavior; tools should operate only on the
+     * provided snapshots which were generated in prior generate_*interval_snapshot
+     * calls.
+     *
+     * Tools should not modify any data in the base \p interval_state_snapshot_t
+     * struct.
+     *
+     * In the parallel mode, this is invoked with the list of shard-local snapshots
```

Comment:
I think it's worth clarifying that in parallel mode this method will be invoked once per shard.  Maybe just change the wording a bit to "...this is invoked on each list of shard-local snapshots...".

Reply:
Done.